core/templates/components/state-directives/outcome-editor/outcome-editor.component.ts

```diff
@@ -71,8 +71,8 @@ export class OutcomeEditorComponent implements OnInit {
 
   isCurrentInteractionLinear(): boolean {
     let interactionId = this.getCurrentInteractionId();
-    return interactionId !== null && INTERACTION_SPECS[
```

Comment:
Earlier `interactionId` was checked for being `falsy` which means both `null` and `undefined` were not allowed. But PR 15620 changed this to only check for `null`. So, the `is_linear` e2e flake was introduced. Later PR #15674 add a ? to suppress the error.

Reply:
In that case would it not make more sense to have explicit `null` and `undefined` checks? Having a boolean check on a number doesn't make sense.

Reply:
`interactionId` is a string. So, possible values are `null`, `undefined` and `''` (empty string). All of these are falsy, so boolean will evaluate this to `false`.

Reply:
sg, thanks for explaining!